src/Maestro/Maestro.MergePolicies/MergePolicy.cs

```diff
@@ -47,6 +49,41 @@ public string Name
 
         public abstract string DisplayName { get; }
 
-        public abstract Task EvaluateAsync(IMergePolicyEvaluationContext context, MergePolicyProperties properties);
+        public abstract Task<MergePolicyEvaluationResult> EvaluateAsync(IPullRequest pr, IRemote darc);
+
+        public Task<MergePolicyEvaluationResult> Pending(string message)
```

Comment:
Why are these async? They aren't virtual, and there is no chance they will do anything Async.

Reply:
I have decided to have this async for two reasons:
1. enable of adding some async into these without changing calling code. For example writing into db.
1. IMergePolicy.EvaluateAsync are sometimes implemented without await in body. In such situation there would have to be `return Task.FromResult(Fail("there was an error"))`. I always optimize code for readability and Task.FromResult lover readability. So I chosed to have lower readability in helper methods, that in core logic methods. This ways IMergePolicy.EvaluateAsync can use `return Fail("error")` or `return await Fail("error")` - both of those perfectly readable

Reply:
We can change code later... making methods return Task just in case is a little weird...  I can't imagine a world where `Fail("Some message")` is async... what sort of scenarios are you envisioning?

Reply:
Possible future cases of await in Fail(""):
- writing results into storage
- sending telemetry data by event hub

But actually the 2 point - increased readability of all MergePolicy.EvaluateAsync implementation - is the main driver here.